File: app/src/main/java/com/ostron/EcoGov/cbwtf/activities/CbwtfLoginActivity.java
package com.ostron.EcoGov.cbwtf.activities;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import android.app.ProgressDialog;
import android.content.ActivityNotFoundException;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.os.Vibrator;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import com.android.volley.AuthFailureError;
import com.android.volley.Request;
import com.android.volley.Response;
import com.android.volley.VolleyError;
import com.android.volley.toolbox.StringRequest;
import com.android.volley.toolbox.Volley;

import com.bumptech.glide.Glide;
import com.google.android.material.snackbar.Snackbar;
import com.google.android.material.textfield.TextInputEditText;
import com.ostron.EcoGov.R;
import com.ostron.EcoGov.activities.PdfViewActivity;
import com.ostron.EcoGov.classes.AppStrings;
import com.ostron.EcoGov.classes.MSP;
import com.ostron.EcoGov.classes.VolleySingleton;

import org.json.JSONObject;

import java.util.HashMap;
import java.util.Map;

public class CbwtfLoginActivity extends AppCompatActivity {

    ImageView headerImg,footerImg;
    TextView guideline_tv,youtube_tv;
    TextInputEditText mobile_et,password_et;

    Vibrator vibrator;
    int loginAs = 0;

    ProgressDialog progressDialog;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_cbwtf_login);

        vibrator = (Vibrator)getSystemService(VIBRATOR_SERVICE);

        progressDialog = new ProgressDialog(this);
        headerImg = findViewById(R.id.cbwtf_login_header_img);
        footerImg = findViewById(R.id.cbwtf_login_bottom_img);
        guideline_tv = findViewById(R.id.cbwtf_guideline_link_tv);
        youtube_tv = findViewById(R.id.cbwtf_youtube_link_tv);
        mobile_et = findViewById(R.id.cbwtf_login_mobile_et);
        password_et = findViewById(R.id.cbwtf_login_password_et);

        Glide.with(this).load(R.drawable.uplogogpb).into(headerImg);
     //   Glide.with(this).load(R.drawable.img_011).into(footerImg);

        loginAs = getIntent().getIntExtra("loginAs",0);

        progressDialog.setCancelable(false);
        progressDialog.setMessage("Please wait...");
        progressDialog.create();

        guideline_tv.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivity(new Intent(CbwtfLoginActivity.this, PdfViewActivity.class));
            }
        });

        youtube_tv.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent appIntent = new Intent(Intent.ACTION_VIEW, Uri.parse("vnd.youtube:j46IAKw6rpI"));
                Intent webIntent = new Intent(Intent.ACTION_VIEW,
                        Uri.parse("http://www.youtube.com/watch?v=j46IAKw6rpI"));
                try {
                    startActivity(appIntent);
                } catch (ActivityNotFoundException ex) {
                    startActivity(webIntent);
                }
                //startActivity(new Intent(CbwtfLoginActivity.this, WebViewActivity.class));
            }
        });

    }

    public void onLogin(View view){

        if (mobile_et.getText().toString().trim().isEmpty()){
            vibrator.vibrate(100);
            Snackbar.make(mobile_et,"Empty number",1000).show();
        }else if (password_et.getText().toString().trim().isEmpty()){
            vibrator.vibrate(100);
            Snackbar.make(password_et,"Empty password",1000).show();
        }else {

            progressDialog.show();

            Map<String,String> map = new HashMap<>();
            map.put("mobile",mobile_et.getText().toString());
            map.put("password",password_et.getText().toString());
            if (loginAs == 0){
                networkRequest(AppStrings.cbwtf_login,map);
            }else {
                networkRequest(AppStrings.hcf_login,map);
            }
        }
    }

    public void networkRequest(String url,Map<String,String> map){
        StringRequest request = new StringRequest(Request.Method.POST, url, new Response.Listener<String>() {
            @Override
            public void onResponse(String response) {

                try {

                    JSONObject jsonObject = new JSONObject(response);

                    if (jsonObject.get("status").toString().equalsIgnoreCase("success")){

                        if (loginAs == 0){
                            String operatorID = jsonObject.getJSONArray("data").getJSONObject(0).get("operator_id").toString();
                            String operatorName = jsonObject.getJSONArray("data").getJSONObject(0).get("name").toString();
                            String operatorAddress = jsonObject.getJSONArray("data").getJSONObject(0).get("address").toString();
                            String operatorMobile = jsonObject.getJSONArray("data").getJSONObject(0).get("mobile").toString();
                            String operatorCbwtfID = jsonObject.getJSONArray("data").getJSONObject(0).get("cbwtf_id").toString();
                            String userPassword = jsonObject.getJSONArray("data").getJSONObject(0).get("password").toString();

                            MSP.getInstance(CbwtfLoginActivity.this).setStringData(AppStrings.userName,operatorName);
                            MSP.getInstance(CbwtfLoginActivity.this).setStringData(AppStrings.userMobile,operatorMobile);
                            MSP.getInstance(CbwtfLoginActivity.this).setStringData(AppStrings.userPassword,userPassword);
                            MSP.getInstance(CbwtfLoginActivity.this).setStringData(AppStrings.userAddress,operatorAddress);
                            MSP.getInstance(CbwtfLoginActivity.this).setStringData(AppStrings.userID,operatorID);
                            MSP.getInstance(CbwtfLoginActivity.this).setStringData(AppStrings.userCbwtfID,operatorCbwtfID);
                            MSP.getInstance(CbwtfLoginActivity.this).setStringData(AppStrings.loginAs,"cbwtf");
                        }else {
                            String operatorID = jsonObject.getJSONArray("data").getJSONObject(0).get("hospital_code").toString();
                            String operatorName = jsonObject.getJSONArray("data").getJSONObject(0).get("name").toString();
                            String operatorAddress = jsonObject.getJSONArray("data").getJSONObject(0).get("address").toString();
                            String operatorMobile = jsonObject.getJSONArray("data").getJSONObject(0).get("mobile").toString();
                            String operatorCbwtfID = jsonObject.getJSONArray("data").getJSONObject(0).get("cbwtf_id").toString();
                            String userPassword = jsonObject.getJSONArray("data").getJSONObject(0).get("password").toString();

                            MSP.getInstance(CbwtfLoginActivity.this).setStringData(AppStrings.userName,operatorName);
                            MSP.getInstance(CbwtfLoginActivity.this).setStringData(AppStrings.userMobile,operatorMobile);
                            MSP.getInstance(CbwtfLoginActivity.this).setStringData(AppStrings.userPassword,userPassword);
                            MSP.getInstance(CbwtfLoginActivity.this).setStringData(AppStrings.userAddress,operatorAddress);
                            MSP.getInstance(CbwtfLoginActivity.this).setStringData(AppStrings.userID,operatorID);
                            MSP.getInstance(CbwtfLoginActivity.this).setStringData(AppStrings.userCbwtfID,operatorCbwtfID);
                            MSP.getInstance(CbwtfLoginActivity.this).setStringData(AppStrings.loginAs,"hcf");
                        }

                        progressDialog.dismiss();

                        Intent intent = new Intent(CbwtfLoginActivity.this, CbwtfDashboardActivity.class);
                        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
                        startActivity(intent);
                        finish();

                    }else {
                        progressDialog.dismiss();
                        vibrator.vibrate(100);
                        Snackbar.make(password_et,"Wrong credential",1000).show();
                    }

                }catch (Exception e){
                    progressDialog.dismiss();
                    e.printStackTrace();
                    Toast.makeText(CbwtfLoginActivity.this, "Something went wrong try again", Toast.LENGTH_SHORT).show();
                }

            }
        }, new Response.ErrorListener() {
            @Override
            public void onErrorResponse(VolleyError error) {
                progressDialog.dismiss();
                Toast.makeText(CbwtfLoginActivity.this, "Something went wrong try again", Toast.LENGTH_SHORT).show();
            }
        }){
            @Nullable
            @Override
            protected Map<String, String> getParams() throws AuthFailureError {
                return map;
            }
        };

        VolleySingleton.getInstance(this).addToRequestQueue(request);
    }
}

File: app/src/main/java/com/appventurez/bmwms/cbwtf/activities/CbwtfLoginActivity.kt
package com.appventurez.bmwms.cbwtf.activities

import android.content.ActivityNotFoundException
import android.content.Intent
import android.net.Uri
import android.os.Build
import android.os.Bundle
import android.os.VibrationEffect
import android.os.Vibrator
import android.os.VibratorManager
import android.util.Log
import android.widget.Toast
import androidx.activity.compose.setContent
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.compose.runtime.LaunchedEffect
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.appventurez.bmwms.activities.PdfViewActivity
import com.appventurez.bmwms.cbwtf.compose.CbwtfLoginScreen
import com.appventurez.bmwms.classes.AppStrings
import com.appventurez.bmwms.classes.MSP
import com.appventurez.bmwms.network.RetrofitClient
import com.appventurez.bmwms.repository.LoginRepository
import com.appventurez.bmwms.viewmodel.LoginViewModel
import java.util.HashMap

class CbwtfLoginActivity : AppCompatActivity() {
    private var TAG = "CbwtfLoginActivity"
    private var vibrator: Vibrator? = null
    private var loginAs = 0

    private val viewModel: LoginViewModel by viewModels {
        object : ViewModelProvider.Factory {
            override fun <T : ViewModel> create(modelClass: Class<T>): T {
                return LoginViewModel(LoginRepository(RetrofitClient.apiService)) as T
            }
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        vibrator = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
            val vibratorManager =
                getSystemService(VIBRATOR_MANAGER_SERVICE) as VibratorManager
            vibratorManager.defaultVibrator
        } else {
            @Suppress("DEPRECATION")
            getSystemService(VIBRATOR_SERVICE) as Vibrator
        }
        
        loginAs = intent.getIntExtra("loginAs", 0)

        setContent {
            val loginResponse = viewModel.loginResponse.value

            LaunchedEffect(loginResponse) {
                loginResponse?.let { response ->
                    if (response.isSuccessful) {
                        val body = response.body()
                        if (body?.status.equals("success", ignoreCase = true) && !body?.data.isNullOrEmpty()) {
                            val data = body!!.data!![0]
                            val msp = MSP.getInstance(this@CbwtfLoginActivity)
                            
                            if (loginAs == 0) {
                                msp.setStringData(AppStrings.userName, data.name)
                                msp.setStringData(AppStrings.userMobile, data.mobile)
                                msp.setStringData(AppStrings.userPassword, data.password)
                                msp.setStringData(AppStrings.userAddress, data.address)
                                msp.setStringData(AppStrings.userID, data.operatorId)
                                msp.setStringData(AppStrings.userCbwtfID, data.cbwtfId)
                                msp.setStringData(AppStrings.loginAs, "cbwtf")
                            } else {
                                msp.setStringData(AppStrings.userName, data.name)
                                msp.setStringData(AppStrings.userMobile, data.mobile)
                                msp.setStringData(AppStrings.userPassword, data.password)
                                msp.setStringData(AppStrings.userAddress, data.address)
                                msp.setStringData(AppStrings.userID, data.hospitalCode)
                                msp.setStringData(AppStrings.userCbwtfID, data.cbwtfId)
                                msp.setStringData(AppStrings.loginAs, "hcf")
                            }

                            val intent = Intent(this@CbwtfLoginActivity, CbwtfDashboardActivity::class.java)
                            intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
                            startActivity(intent)
                            finish()
                        } else {
                            vibrate(100)
                            Toast.makeText(this@CbwtfLoginActivity, "Wrong credential", Toast.LENGTH_SHORT).show()
                            viewModel.resetLoginResponse()
                        }
                    } else {
                        Toast.makeText(this@CbwtfLoginActivity, "Something went wrong try again", Toast.LENGTH_SHORT).show()
                        viewModel.resetLoginResponse()
                    }
                }
            }

            CbwtfLoginScreen(
                isLoading = viewModel.isLoading.value,
                onLoginClick = { mobile, password ->
                    onLoginClick(mobile, password)
                },
                onGuidelineClick = {
                    startActivity(Intent(this, PdfViewActivity::class.java))
                },
                onYoutubeClick = {
                    openYoutube()
                }
            )
        }
    }

    private fun openYoutube() {
        val appIntent = Intent(Intent.ACTION_VIEW, Uri.parse("vnd.youtube:j46IAKw6rpI"))
        val webIntent = Intent(Intent.ACTION_VIEW, Uri.parse("http://www.youtube.com/watch?v=j46IAKw6rpI"))
        try {
            startActivity(appIntent)
        } catch (ex: ActivityNotFoundException) {
            Log.d(TAG, "ActivityNotFoundException "+ ex.message)
            startActivity(webIntent)
        }
    }

    private fun vibrate(duration: Long) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            vibrator?.vibrate(VibrationEffect.createOneShot(duration, VibrationEffect.DEFAULT_AMPLITUDE))
        } else {
            @Suppress("DEPRECATION")
            vibrator?.vibrate(duration)
        }
    }

    private fun onLoginClick(mobile: String, password: String) {
        if (mobile.trim().isEmpty()) {
            vibrate(100)
            Toast.makeText(this, "Empty number", Toast.LENGTH_SHORT).show()
        } else if (password.trim().isEmpty()) {
            vibrate(100)
            Toast.makeText(this, "Empty password", Toast.LENGTH_SHORT).show()
        } else {
            val map = HashMap<String, String>()
            map["mobile"] = mobile
            map["password"] = password
            val url = if (loginAs == 0) AppStrings.cbwtf_login else AppStrings.hcf_login
            viewModel.login(url, map)
        }
    }
}
